File: csharp/GardenCatalogue-TestDesign-Kata/GardenCatalogueTest/PlantPrinter.cs
using System.Text;
using System.Globalization;

namespace GardenCatalogue;

public class PlantPrinter
{
    public void Print(StringBuilder sb, Plant plant, string indent = "")
    {
        void AppendIfNotEmpty(string label, string? value)
        {
            if (!string.IsNullOrEmpty(value))
                sb.AppendLine($"{indent}{label}: {value}");
        }

        sb.AppendLine($"{indent}Name: {plant.Name}");
        AppendIfNotEmpty("Latin Name", plant.LatinName);
        AppendIfNotEmpty("Article #", plant.ArticleNumber);
        
        if (plant.Type != PlantType.Flower)
            sb.AppendLine($"{indent}Type: {plant.Type}");

        if (plant.BloomPeriod.Months.Any())
            sb.AppendLine($"{indent}Blooms: {string.Join(", ", plant.BloomPeriod.Months)}");

        sb.AppendLine($"{indent}Max Height: {plant.MaxHeight.ToString("F1", CultureInfo.InvariantCulture)}m");
        sb.AppendLine($"{indent}Soil: {plant.Soil}");
        sb.AppendLine($"{indent}Light: {plant.Light}");

        if (plant.Properties.Any())
        {
            sb.AppendLine($"{indent}Properties:");
            foreach (var (key, value) in plant.Properties)
                sb.AppendLine($"{indent}  - {key}: {value}");
        }
    }
}

File: python/tests/plant_printer.py
from garden_catalogue.models import Plant, PlantType

class PlantPrinter:
    def print(self, plant: Plant, indent: str = "") -> str:
        lines = [f"{indent}Name: {plant.name}"]
        if plant.latin_name:
            lines.append(f"{indent}Latin Name: {plant.latin_name}")
        if plant.article_number:
            lines.append(f"{indent}Article #: {plant.article_number}")
        
        if plant.type != PlantType.Flower:
            lines.append(f"{indent}Type: {plant.type.name}")

        if plant.bloom_period.months:
            months_str = ", ".join(m.name for m in plant.bloom_period.months)
            lines.append(f"{indent}Blooms: {months_str}")

        lines.append(f"{indent}Max Height: {plant.max_height:.1f}m")
        lines.append(f"{indent}Soil: {plant.soil.name}")
        lines.append(f"{indent}Light: {plant.light.name}")

        if plant.properties:
            lines.append(f"{indent}Properties:")
            for key, value in sorted(plant.properties.items()):
                lines.append(f"{indent}  - {key}: {value}")
        
        return "\n".join(lines) + "\n"
